### ISSUE TITLE
Pylint found no problems

### ISSUE THREAD

**Issue by tispratik** [leinardi/pylint-pycharm#69]
### Step 1: Are you in the right place?
- [x] I have verified there are no duplicate active or recent bugs, questions, or requests
- [x] I have verified that I am using the latest version of the plugin.

### Step 2: Describe your environment
 - Plugin version: `0.12.2`
 - PyCharm/IDEA version: `2019.1 CE`
 - Pylint version: `2.5.2`

### Step 3: Describe the problem:
When i run pylint from terminal, it reports errors. But from Intellij, it does not. 

#### Steps to reproduce:

  1. Open a python file
  2. Run the pylint plugin
  3. "Pylint found no problems" is reported

#### Observed Results:

  * Intellij logfile shows no errors
  
#### Expected Results:

  * Errors should have been reported
  * I also have MyPy plugin installed. It is correctly reporting errors.
  * I am using virtual environment using `venv`
  * Restarting Intellij did not help

#### Relevant Code:
<img width="1014" alt="Screen Shot 2020-05-27 at 7 31 26 PM" src="https://user-images.githubusercontent.com/80631/83092033-5b226f80-a051-11ea-912a-08506612d13e.png">
<img width="1680" alt="Screen Shot 2020-05-27 at 7 32 01 PM" src="https://user-images.githubusercontent.com/80631/83092037-5e1d6000-a051-11ea-9301-8f1845038cdb.png">
<img width="1207" alt="Screen Shot 2020-05-27 at 7 34 57 PM" src="https://user-images.githubusercontent.com/80631/83092038-5f4e8d00-a051-11ea-8a50-5a9acb73dc12.png">
<img width="859" alt="Screen Shot 2020-05-27 at 7 41 51 PM" src="https://user-images.githubusercontent.com/80631/83092447-29f66f00-a052-11ea-9cf0-df373dca4c21.png">

**Comment by tispratik** [comment]
The issue is related to loading of **pylintrc**. When i remove the "/Users/pratik/Projects/main/python/pylintrc" path, then it works (but does not use my pylintrc, uses the default one)

**Comment by tispratik** [comment]
After removing pylintrc path, it works.
<img width="1354" alt="Screen Shot 2020-05-27 at 7 45 39 PM" src="https://user-images.githubusercontent.com/80631/83092710-b012b580-a052-11ea-84bf-afe99d66201f.png">

**Comment by tispratik** [comment]
My pylintrc contains the following line, which causes the issue. If i remove the below line, it works:

```
init-hook="from pylint.config import find_pylintrc; import os, sys; sys.path.append(os.path.dirname(find_pylintrc()))"
```

**Comment by liry** [comment]
I have the same problem but even without pylintrc path set :(
(IDEA 2020.1.2 Ultimate edition, Pylint plugin 0.12.2, Python plugin 201.78846.85)

Idea log just says:
`2020-06-19 13:44:48,253 [1705284]   INFO - di.pycharm.pylint.PylintPlugin - Scanning current file(s).`
and finds nothing even though cmd line version does find some issues

**Comment by tad3j** [comment]
I'm experiencing the same issue where I just can't get pylint plugin to produce any output, but calling pylint from command line works.
OS: Linux Mint
Plugin version: 0.12.2
PyCharm version: 2020.1.3 Ultimate
Pylint version: 3.0.0-a4

I just switched from Windows to Linux and strangely the plugin works there (config sahred via VCS) and even works on friends macOS, so it seems like issue may be specific to Linux.

I'm not sure how to debug it since plugin produces no output (either success or error).

Removing .pylintrc from config doesn't help, hardcoding pylint path instead of plugin autodetecting it also doesn't change anything, so I'm kind of stuck.

**Comment by maciejmatuszak** [comment]
I have been encountered this as well.
In general I notice that most "outdated" config items or missing plugins will cause the plugin to "quietly fail" meaning it will just display "Pylint found no problems" 

To replicate the issue add `disable=no-init`

**The symptoms:**
in pylint commandline output:
```shell
************* Module <path>/pylintrc
pylintrc:1:0: R0022: Useless option value for '--disable', 'no-init' was removed from pylint, see https://github.com/PyCQA/pylint/issues/2409. (useless-option-value)
```
entry in the `idea.log`:
```log
2022-10-05 12:57:20,917 [9021467]   INFO - #c.l.p.p.p.ProcessResultsThread - Could not find mapping for file: pylintrc in {<path redacted>/unit_tests/test_dvc_data_set.py=PyFile:test_dvc_data_set.py}
```

The workaround for me is to cleanup the config
In long term this plugin need to detect this problem and report it instead of "quietly ignoring" it.

**Comment by kannes** [comment]
I get the same behaviour with the official example pylintrc https://github.com/PyCQA/pylint/blob/f860e3fe1c03be81e16732123a1e0cdb247db47f/examples/pylintrc

When scanning the file outside PyCharm and thus not with the plugin, multiple problems are shown, in my case no `R*` problems, no "Useless option ..." ones.

**Comment by bandophahita** [comment]
I have a slight variation on this same issue.
PyCharm 2022.3.3 (Professional Edition)
Build #PY-223.8836.43, built on March 10, 2023
macOS 13.2.1

I can get pylint to work in one project but not another.

The project where pylint plugin detects error is using python 3.10
Whereas the project that doesnt is using 3.11

The project where it doesn't detect errors in the plugin, works when using commandline.
They have slightly different pylintrc files.   I tried copying the "working" pylintrc to the other project.  Didn't seem to have any impact.

**Comment by vt-cwalker** [comment]
If anyone else finds this thread, my issue was also an invalid linting key that caused my issue. I had `max-complexity` in my `.pylintrc` file and once i removed it, pylint started working again in PyCharm

**Comment by leinardi** [comment]
Hi everyone,  

A new developer, `@szabope`, has created a full rewrite of this PyLint plugin. The new plugin is actively maintained and addresses many of the issues found here.  

You can find the new plugin on the JetBrains Plugin Marketplace: [PyLint Plugin](https://plugins.jetbrains.com/plugin/26358-pylint/versions/stable/668924) 
The source code is available here: [GitHub Repository](https://github.com/szabope/pylint-pycharm-plugin)

Please give it a try, as the old plugin will soon be archived.  

Thank you!
